File: MostContainerApproach.js
const area=(array)=>
{
    //max Area
    maxArea=0
    //left pointer
    left=0
    //right pointer
    right=array.length-1
    while(left<right)
    {
        //area=min(array[left],array[right])*(j-1)
        height=Math.min(array[left],array[right])
        width=right-left
        //max Area 
        maxArea=Math.max(maxArea,height*width)
        //shifting the pointers
        if(array[left]<array[right]) left+=1
        else right-=1

    }
    return maxArea
}
array=[3,7,5,6,8,4]
console.log(area(array))

File: MostContainerApproach.py
class MaxArea:
    #method to find maximum area
    def findingArea(array):
        #var declaration
        maxArea=0
        #letft pointer
        left=0
        #right pointer
        right=len(array)-1
        #stopping condition
        while left<right:
            #area=min(array[left],array[right])*(j-i)
            #finding the min between two arrays
            height=min(array[left],array[right])
            #finding the width difference of indexes
            width=right-left
            #max area 
            maxArea=max(maxArea,height*width)
            #shifying the pointers according to min element
            if array[left]<array[right]:left+=1
            else: right-=1
        return maxArea
array=[3,7,5,6,8,4]
print(MaxArea.findingArea(array))